pkg/gatherers/clusterconfig/gather_monitoring_pvs.go

```diff
@@ -0,0 +1,136 @@
+package clusterconfig
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/openshift/insights-operator/pkg/record"
+	"github.com/openshift/insights-operator/pkg/utils"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	coreV1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"sigs.k8s.io/yaml"
+)
+
+// GatherMonitoringPVs Collects Persistent Volumes from openshift-monitoring namespace
+// which matches with ConfigMap configuration yaml
+//
+// ### API Reference
+// - https://github.com/kubernetes/client-go/blob/master/kubernetes/typed/core/v1/configmap.go
+// - https://github.com/kubernetes/client-go/blob/master/kubernetes/typed/core/v1/persistentvolume.go
+//
+// ### Sample data
+// - docs/insights-archive-sample/config/persistentvolumes/monitoring-persistent-volume.json
+//
+// ### Location in archive
+// - `config/persistentvolumes/{persistent_volume_name}.json`
+//
+// ### Config ID
+// `clusterconfig/monitoring_persistent_volumes`
+//
+// ### Released version
+// - 4.14
+//
+// ### Changes
+// None
+func (g *Gatherer) GatherMonitoringPVs(ctx context.Context) ([]record.Record, []error) {
+	kubeClient, err := kubernetes.NewForConfig(g.gatherProtoKubeConfig)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	mg := MonitoringPVGatherer{client: kubeClient.CoreV1()}
+
+	name, err := mg.getVolumeClaimName(ctx)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	return mg.gather(ctx, name)
+}
+
+type MonitoringPVGatherer struct {
+	client coreV1.CoreV1Interface
+}
+
+// getVolumeClaimName returns VolumeClaim name used on prometheus configuration
+// as it's described on the ConfigMap
+// or an error from retrieving that information
+func (mg MonitoringPVGatherer) getVolumeClaimName(ctx context.Context) (string, error) {
+	const CMO = "cluster-monitoring-config"
+	const NAMESPACE = "openshift-monitoring"
+	const CONFIG = "config.yaml"
+
+	cm, err := mg.client.ConfigMaps(NAMESPACE).Get(ctx, CMO, metaV1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	rawData, exists := cm.Data[CONFIG]
+	if !exists {
+		return "", fmt.Errorf("no %s data on %s ConfigMap", CONFIG, CMO)
+	}
+
+	name, err := mg.unmarshalVCTemplateName(rawData)
+	if err != nil {
+		return "", err
+	}
+
+	return name, nil
+}
+
+// unmarshalVCTemplateName returns VC name from configuration raw data (yaml format)
+// or an error if the raw data is not unmarshalable or it lacks the default path
+func (mg MonitoringPVGatherer) unmarshalVCTemplateName(raw string) (string, error) {
+	var defaultPath = []string{"prometheusK8s", "volumeClaimTemplate", "metadata", "name"}
```

Comment:
Is this actually right? I am looking at https://docs.openshift.com/container-platform/4.12/monitoring/configuring-the-monitoring-stack.html and I guess it should be `prometheusK8s`, `volumeClaimTemplate`, `spec`,`storageClassName`

Reply:
Nope....I was completely wrong. I didn't get that the PVC is created (by prometheus-operator or CMO probably) automatically and is/are called `prometheus-k8s-db-prometheus-k8s-0`, `prometheus-k8s-db-prometheus-k8s-1` by default. I apologize for my ignorance.